src/rgw/rgw_sal_motr.cc

```diff
@@ -1910,7 +1919,8 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
     }
     // Call `cb` to process returned data.
     ldpp_dout(dpp, 20) << "MotrObject::read_mobj(): call cb to process data" << dendl;
-    cb->handle_data(bl, 0, actual);
+    cb->handle_data(bl, bloff, actual);
```

Comment:
`bloff` needs to calculated correctly based on which block is being read. for example - if `start` lies somewhere in the second block then `bloff = start - block_start_off` (read bl from custom offset) and for the next block onwards the `bloff = 0` (read bl from the start)

Reply:
In bloff we are storing the value of 'off' which is coming from function parameter, and we will call handle_data function when bloff is present in that block only, so no need of calculation of start - block_start_off, it will get the data from correct offset only, and after the handle_data function call we are setting the bloff to 0.

Reply:
bloff should always be relative offset from the start of the block, once the block is present in bl. Lets discuss this more offline.